apps/scratch/src/pages/tests/Edit.spec.tsx

```diff
@@ -0,0 +1,142 @@
+import { readFileSync } from "fs";
+import { resolve } from "path";
+import { test, expect } from "playwright-test-coverage";
+
+// eslint-disable-next-line no-undef
+const testTask = readFileSync(resolve(__dirname, "test-task.zip"));
+
+declare global {
+  interface Window {
+    postedMessages: { message: unknown; options: unknown }[];
+  }
+}
+
+test.describe("/edit/taskId", () => {
+  test.beforeEach(async ({ page, baseURL }) => {
+    page.on("framenavigated", async () =>
+      page.evaluate(() => {
+        window.postedMessages = [];
+
+        // @ts-expect-error - we mock the parent window
+        window.parent = {
+          postMessage: (message, options) => {
+            window.postedMessages.push({ message, options });
+          },
+        };
+      }),
+    );
+
+    await page.goto(`${baseURL!}/edit/some-task-id`);
+
+    await page.waitForSelector("#root");
+  });
+
+  test("can get height via window.postMessage", async ({ page }) => {
+    await page.evaluate(() => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const event = new Event("message") as any;
+
+      event.source = window.parent;
+      event.data = { id: 0, type: "request", procedure: "getHeight" };
+      window.dispatchEvent(event);
+    });
+
+    await page.waitForFunction(() => window.postedMessages.length > 0);
+
+    const messages = await page.evaluate(() => window.postedMessages);
+
+    expect(messages).toHaveLength(1);
+
+    expect(messages[0].message).toEqual({
+      id: 0,
+      type: "response",
+      procedure: "getHeight",
+      result: expect.any(Number),
+    });
+  });
+
+  test("can get task via window.postMessage", async ({ page }) => {
+    await page.evaluate(() => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const event = new Event("message") as any;
+
+      event.source = window.parent;
+      event.data = { id: 0, type: "request", procedure: "getTask" };
+      window.dispatchEvent(event);
+    });
+
+    await page.waitForFunction(() => window.postedMessages.length > 0);
+
+    const messages = await page.evaluate(() => window.postedMessages);
+
+    expect(messages).toHaveLength(1);
+
+    expect(messages[0].message).toEqual({
+      id: 0,
+      type: "response",
+      procedure: "getTask",
+      // blobs cannot be transferred, see https://github.com/puppeteer/puppeteer/issues/3722
+      result: {},
```

Comment:
That same ticket seems to say that you can with some ugly workaround: https://github.com/puppeteer/puppeteer/issues/3722#issuecomment-1656736694

Reply:
Yes I saw that but unfortunately it then becomes very difficult to assert anything useful on that other than "is a string". Do you think it is still worth it?